docs/changelog/127252.yaml

```diff
@@ -0,0 +1,5 @@
+pr: 127252
+summary: New per-project only settings can be defined and used by components
```

Comment:
This isn't designed to be user-facing, so I wouldn't flag this as a feature for inclusion in release notes.